packages/api/src/sequelize/migrations/2023-07-25_01_create-docref-mapping.ts

```diff
@@ -0,0 +1,53 @@
+import { DataTypes } from "sequelize";
+import type { Migration } from "..";
+import * as shared from "../migrations-shared";
+
+const tableName = "docref_mapping";
+
+// Use 'Promise.all' when changes are independent of each other
+export const up: Migration = async ({ context: queryInterface }) => {
+  await queryInterface.sequelize.transaction(async transaction => {
+    await shared.createTable(
+      queryInterface,
+      tableName,
+      {
+        id: {
+          type: DataTypes.STRING,
+          primaryKey: true,
+        },
+        externalId: {
+          type: DataTypes.STRING,
+          field: "external_id",
+        },
+        cxId: {
+          type: DataTypes.UUID,
+          field: "cx_id",
+        },
+        patientId: {
+          type: DataTypes.STRING,
+          field: "patient_id",
+        },
+        source: {
+          type: DataTypes.STRING,
+        },
+      },
+      {
+        transaction,
+        addVersion: true,
+        uniqueKeys: {
+          external_unique: { fields: ["external_id", "patient_id", "cx_id", "source"] },
+        },
+      }
+    );
+    await queryInterface.addIndex(tableName, ["cx_id", "patient_id", "source", "external_id"], {
+      name: tableName + "_alternative_unique_idx",
+      transaction,
+    });
```

Comment:
So we can query doc ref mappings by cx, cx+patient, cx+patient+source

Reply:
NDB right now - just want to flag we may have to re-evaluate this index down the line for perf reasons - we'll be writing to this table a lot, and the index will need to be re-computed on each write

Reply:
Yeah, we likely will need to create/update lots of indexes as we grow. I assumed we'll likely need to query/count records here by cxId, so ended up creating it. Actually, most tables with cxId should have an index with that column as the first one on the index.

Good that you brought it up, because it didn't make sense to have external_id on this index - if it comes to having external_id as a filter, surely the DB will use the unique constraint's index to hit the record directly.